### PR TITLE
Fix NVDA not reading in Geekbench 6

### LINKED ISSUE #16251: NVDA does not read in the Geekbench 6 utility
### Steps to reproduce:
1. download and install geekbench
2. open the application and try to press alt to access the ribbon
### Actual behavior:

 NVDA will not read the program and will not read options in ribbon. Jaws is able to work with the application. The following errors generate in log

ERROR - core.CorePump.Notify (04:18:35.343) - MainThread (5648):
errors in this core pump cycle
Traceback (most recent call last):
  File "core.pyc", line 837, in Notify
  File "IAccessibleHandler\__init__.pyc", line 1063, in pumpAll
  File "IAccessibleHandler\__init__.pyc", line 725, in processFocusWinEvent
  File "IAccessibleHandler\__init__.pyc", line 560, in winEventToNVDAEvent
  File "UIAHandler\__init__.pyc", line 1242, in isUIAWindow
  File "UIAHandler\__init__.pyc", line 1147, in _isUIAWindowHelper
  File "baseObject.pyc", line 41, in __get__
  File "appModuleHandler.pyc", line 526, in _get_productName
  File "appModuleHandler.pyc", line 521, in _setProductInfo
  File "appModuleHandler.pyc", line 480, in _getExecutableFileInfo
  File "fileUtils.pyc", line 67, in funcWrapper
  File "fileUtils.pyc", line 82, in getFileVersionInfo
RuntimeError: No version information
ERROR - core.CorePump.Notify (04:18:35.354) - MainThread (5648):
errors in this core pump cycle
Traceback (most recent call last):
  File "core.pyc", line 837, in Notify
  File "IAccessibleHandler\__init__.pyc", line 1063, in pumpAll
  File "IAccessibleHandler\__init__.pyc", line 725, in processFocusWinEvent
  File "IAccessibleHandler\__init__.pyc", line 560, in winEventToNVDAEvent
  File "UIAHandler\__init__.pyc", line 1242, in isUIAWindow
  File "UIAHandler\__init__.pyc", line 1147, in _isUIAWindowHelper
  File "baseObject.pyc", line 41, in __get__
  File "appModuleHandler.pyc", line 526, in _get_productName
  File "appModuleHandler.pyc", line 521, in _setProductInfo
  File "appModuleHandler.pyc", line 480, in _getExecutableFileInfo
  File "fileUtils.pyc", line 67, in funcWrapper
  File "fileUtils.pyc", line 82, in getFileVersionInfo
RuntimeError: No version information
ERROR - comtypes._comobject.call_without_this (04:18:35.364) - Dummy-5 (2776):
Exception in IUIAutomationFocusChangedEventHandler.HandleFocusChangedEvent implementation:
Traceback (most recent call last):
  File "comtypes\_comobject.pyc", line 154, in call_without_this
  File "UIAHandler\__init__.pyc", line 843, in IUIAutomationFocusChangedEventHandler_HandleFocusChangedEvent
  File "UIAHandler\__init__.pyc", line 1399, in isNativeUIAElement
  File "UIAHandler\__init__.pyc", line 1242, in isUIAWindow
  File "UIAHandler\__init__.pyc", line 1147, in _isUIAWindowHelper
  File "baseObject.pyc", line 41, in __get__
  File "appModuleHandler.pyc", line 526, in _get_productName
  File "appModuleHandler.pyc", line 521, in _setProductInfo
  File "appModuleHandler.pyc", line 480, in _getExecutableFileInfo
  File "fileUtils.pyc", line 67, in funcWrapper
  File "fileUtils.pyc", line 82, in getFileVersionInfo
RuntimeError: No version information
ERROR - core.CorePump.Notify (04:18:35.372) - MainThread (5648):
errors in this core pump cycle
Traceback (most recent call last):
  File "core.pyc", line 837, in Notify
  File "IAccessibleHandler\__init__.pyc", line 1059, in pumpAll
  File "IAccessibleHandler\__init__.pyc", line 650, in processGenericWinEvent
  File "IAccessibleHandler\__init__.pyc", line 560, in winEventToNVDAEvent
  File "UIAHandler\__init__.pyc", line 1242, in isUIAWindow
  File "UIAHandler\__init__.pyc", line 1147, in _isUIAWindowHelper
  File "baseObject.pyc", line 41, in __get__
  File "appModuleHandler.pyc", line 526, in _get_productName
  File "appModuleHandler.pyc", line 521, in _setProductInfo
  File "appModuleHandler.pyc", line 480, in _getExecutableFileInfo
  File "fileUtils.pyc", line 67, in funcWrapper
  File "fileUtils.pyc", line 82, in getFileVersionInfo
RuntimeError: No version information

### Expected behavior:

NVDA reads the program and reads options in ribbon

#### NVDA installed/portable/running from source:
installed
#### NVDA version:
NVDA version alpha-31229,74a19985
#### Windows version:
windows 11 23h2 (OS Build 22631.3235)
#### Name and version of other software in use when reproducing the issue:
Geekbench 6
#### Does the issue still occur after restarting your computer?
yes
#### Have you tried any other versions of NVDA? If so, please report their behaviors.
no
#### If NVDA add-ons are disabled, is your problem still occurring?
yes
#### Does the issue still occur after you run the COM Registration Fixing Tool in NVDA's tools menu?
yes

### PR DESCRIPTION
**Pull request by mzanm**
### Link to issue number:
Fixes #16251 

### Summary of the issue:
NVDA does not read in Geekbench 6 and plays an error sound.

### Description of user facing changes
Geekbench can be used with NVDA.

### Description of development approach
See diff and linked issue.

### Testing strategy:
Tested that Geekbench can be used with NVDA and no errors appear in the log.

### Known issues with pull request:
None.

### Code Review Checklist:

- [x] Documentation:
  - Change log entry
  - User Documentation
  - Developer / Technical Documentation
  - Context sensitive help for GUI changes
- [x] Testing:
  - Unit tests
  - System (end to end) tests
  - Manual testing
- [x] UX of all users considered:
  - Speech 
  - Braille
  - Low Vision
  - Different web browsers
  - Localization in other languages / culture than English
- [x] API is compatible with existing add-ons.
- [x] Security precautions taken.

### CODE REVIEW

**Comment by CyrilleB79** [comment]
An item in the bugfix section of the change log would also be desirable. Thanks.

File: source/UIAHandler/__init__.py [hunk 1]
```diff
@@ -1148,8 +1148,8 @@ def _isUIAWindowHelper(self, hwnd: int, isDebug=False) -> bool:  # noqa: C901
 			try:
 				isOfficeApp = appModule.productName.startswith(("Microsoft Office", "Microsoft Outlook"))
 				isOffice2013OrOlder = int(appModule.productVersion.split(".")[0]) < 16
-			except Exception:
-				log.debugWarning("Unable to parse productName or productVersion, version information likely unavailable", exc_info=True)
+			except RuntimeError:
+				log.debugWarning("Failed parsing productName / productVersion, versionInfo likely missing", exc_info=True)
```
Comment (seanbudd) [inline source/UIAHandler/__init__.py:1152]: `versionInfo` is a module in NVDA. I don't think this abbreviation makes sense here. If you want to fix the lint, break the content into new lines.

```suggestion
				log.debugWarning(
					"Failed parsing productName / productVersion, version information likely missing",
					exc_info=True
				)
```

File: user_docs/en/changes.t2t [hunk 2]
```diff
@@ -62,6 +62,7 @@ What's New in NVDA
   - ``alt+upArrow`` is now mapped to ``dot2+dot3+dot7+space``
   - ``alt+downArrow`` is now mapped to ``dot5+dot6+dot7+space``
   -
+- Fixed a bug causing NVDA to  fail to  read the ribbon and options within Geekbench. (#16251, @mzanm)
```
Comment (seanbudd) [inline user_docs/en/changes.t2t:65]:
```suggestion
- Fixed a bug causing NVDA to fail to read the ribbon and options within Geekbench. (#16251, @mzanm)
```

**Review by seanbudd — APPROVED** [review]

File: source/UIAHandler/__init__.py [hunk 3]
```diff
@@ -1144,8 +1144,17 @@ def _isUIAWindowHelper(self, hwnd: int, isDebug=False) -> bool:  # noqa: C901
 			# Using IAccessible for NetUIHWND controls causes focus changes not to be reported
 			# when the ribbon is collapsed.
 			# Testing shows that these controls emits proper events but they are ignored by NVDA.
-			isOfficeApp = appModule.productName.startswith(("Microsoft Office", "Microsoft Outlook"))
-			isOffice2013OrOlder = int(appModule.productVersion.split(".")[0]) < 16
+			# this is not necessarily an office app, or an app with version information, for example geekbench 6.
+			try:
+				isOfficeApp = appModule.productName.startswith(("Microsoft Office", "Microsoft Outlook"))
+				isOffice2013OrOlder = int(appModule.productVersion.split(".")[0]) < 16
+			except RuntimeError:
```
Comment (seanbudd) [inline source/UIAHandler/__init__.py:1151]: Shouldn't this warning be here?

```suggestion
			try:
				isOfficeApp = appModule.productName.startswith(("Microsoft Office", "Microsoft Outlook"))
				isOffice2013OrOlder = int(appModule.productVersion.split(".")[0]) < 16
			except RuntimeError:
				# this is not necessarily an office app, or an app with version information, for example geekbench 6.
```

**Pull request merged**